File: benchmarking/count.hs
import System.CPUTime (getCPUTime)

approxPi :: Int -> Double
approxPi n = 4 * sum [((-1) ** fromIntegral k) / (2 * fromIntegral k + 1) | k <- [0..n]]

main :: IO ()
main = do
    let iterations = 1000000
    start <- getCPUTime
    let piApprox = approxPi iterations
    end <- getCPUTime
    let timeMs = fromIntegral (end - start) / (10^9)

    putStrLn $ "Temps d'exécution en Haskell : " ++ show timeMs ++ " ms"

-- merci chatgpt pour l'haskell tu gères

File: benchmarking/count.py
import time

def calculer_pi(nb_iterations):
    pi = 0.0
    for n in range(nb_iterations):
        pi += (-1) ** n / (2 * n + 1)
    return pi * 4

start = time.time()
pi_approx = calculer_pi(1_000_000)
end = time.time()

print(f"Temps d'exécution en Python : {(end - start) * 1000:.2f} ms")
